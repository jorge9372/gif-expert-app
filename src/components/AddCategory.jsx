import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const onSubmit = (e) => {
        e.preventDefault();

        /* Aca lo que se hace es convertir el string a un string sin espacios en blanco
        si la longitud de dicho string es menor o igual a 1 no continua ejecutando el
        resto de la funcion para no agregar elemntos sin contenido, en este caso minimo con
        dos letras
        */
        if (inputValue.trim().length <= 1) return;

        onNewCategory(inputValue.trim());

        // Para vaciar la caja de texto
        setInputValue('');
    };

    return (
        <form className="search-bar" onSubmit={onSubmit} aria-label='form'>
            <input
                type="text"
                value={inputValue}
                placeholder="Buscar gifs"
                onChange={onInputChange}
            />
            <button>Agregar</button>
        </form>
    );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
};
