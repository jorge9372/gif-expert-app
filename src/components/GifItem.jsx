import PropTypes from 'prop-types'  

export const GifItem = ({ title, url }) => {
    

    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
};


/*
    AÃ±adir PropTypes... npm install
        title obligatorio
        url obligatorio

    Evaluar el snapshot

*/


GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired    
}
