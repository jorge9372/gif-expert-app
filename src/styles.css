* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgb(234, 234, 234);
}

body {
    padding: 60px;
}

input {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 100%;
}

button {
    padding: 10px 10px;
    background-color: cornflowerblue;
    color: whitesmoke;
    font-size: 1.5rem;
    border: solid 1px mediumpurple;
    border-radius: 5px;
    box-shadow: 2px 2px 0px rgba(0, 0, 0, 0.05);
    transition: 10ms;
}

button:hover {
    box-shadow: 2px 2px 0px rgba(0, 0, 0, 0.2);
}

button:active {
    border: transparent;
}

h1 {
    font-size: 3rem;
}

h2 {
    font-size: 2.5rem;
}

h3 {
    font-size: 2rem;
    margin-bottom: 5px;
}

.card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}

.card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
    display: flex;
    flex-direction: column;
    height: 0%;
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
}

.card p {
    background-color: white;
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
}

.card img {
    width: 100%;
}

.search-bar {
    display: flex;
    flex-direction: row;
    justify-content: center;
    gap: 5px;
    width: 100%;
}
