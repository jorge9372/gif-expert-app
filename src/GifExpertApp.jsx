import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Rugby']);

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;
        setCategories([newCategory, ...categories]);
    };

    //TODO: Implementar funcion para eliminar elemento
    const onDelete = (category) => {
        setCategories(categories.filter((element) => element !== category));
    };

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory onNewCategory={onAddCategory} />

            {categories.map((category) => (
                <GifGrid
                    onDelete={onDelete}
                    category={category}
                    key={category}
                />
            ))}
        </>
    );
};
