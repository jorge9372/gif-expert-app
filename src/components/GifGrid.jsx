import { GifItem } from './GifItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

import PropTypes from 'prop-types';

export const GifGrid = ({ category, onDelete }) => {
    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <div className="card-grid-header">
                <h2>{category}</h2>
                <button onClick={() => {onDelete(category)}}>Eliminar</button>
            </div>
            {isLoading && <h3>Cargando...</h3>}
            <div className="card-grid">
                {images.map((image) => (
                    <GifItem key={image.id} {...image} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
