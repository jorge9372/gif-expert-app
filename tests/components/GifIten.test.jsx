import { render } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('Pruebas en <GetItem />', () => {
    const title = 'Dragon Ball';
    const url = 'https://api.giphy.com';
    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });
});
