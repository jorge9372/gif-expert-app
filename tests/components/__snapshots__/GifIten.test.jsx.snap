// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GetItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://api.giphy.com"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
